rasterio/vfs.py

```diff
@@ -14,13 +16,17 @@ def parse_path(path, vfs=None):
     """Parse a file path or Apache VFS URL into its parts."""
     archive = scheme = None
     if vfs:
-        parts = vfs.split("://")
-        scheme = parts.pop(0) if parts else None
-        archive = parts.pop(0) if parts else None
+        parts = parse(vfs)
+        scheme = parts['scheme']
+        archive = parts['path']
+        if parts['authority'] and parts['authority'] != 'localhost':
+            archive = parts['authority'] + archive
     else:
-        parts = path.split("://")
-        path = parts.pop() if parts else None
-        scheme = parts.pop() if parts else None
+        parts = parse(path)
+        scheme = parts['scheme']
+        path = parts['path']
+        if parts['authority'] and parts['authority'] != 'localhost':
+            path = parts['authority'] + path
```

Comment:
@geowurster I've got an idea about also supporting AWS ARNs, but it looks like `urlparse()` handles them well enough.

```
>>> urlparse('arn:aws:s3:::my_corporate_bucket/exampleobject.png')
ParseResult(scheme='arn', netloc='', path='aws:s3:::my_corporate_bucket/exampleobject.png', params='', query='', fragment='')
```

I'll switch back.